File: 1047-remove-all-adjacent-duplicates-in-string/1047-remove-all-adjacent-duplicates-in-string.js
/**
 * @param {string} s
 * @return {string}
 */
const removeDuplicates = function(s) {
    if (!s) return '';
    const stack = [s[0]];
    for (let i = 1; i < s.length; i++) {
        stack[stack.length - 1] === s[i] ?
            stack.pop() :
            stack.push(s[i]);
    }
    return stack.join('');
};

File: 1047-remove-all-adjacent-duplicates-in-string/1047-remove-all-adjacent-duplicates-in-string.rb
# @param {String} s
# @return {String}
def remove_duplicates(s)
    stack = [s[0]]
    
    s[1..].each_char do |c|
        if c == stack.last
            stack.pop
        else
            stack << c
        end
    end
    
    stack.join
end